src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -505,7 +513,7 @@ class FunctionDefinitionGrammar(BaseSegment):
     type = "function_definition"
     match_grammar = Sequence(
         "AS",
-        OneOf(Ref("QuotedLiteralSegment"), Ref("DollarQuotedLiteralSegment")),
```

Comment:
`DollarQuotedLiteralSegment` isn't anywhere else in the code base. We used to have it in Postgres, but was since combined with `QuotedLiteralSegment` to simplify the dialect. This is probably just a holdover from when Snowflake inherited from Postgres.